app/src/store/data.ts

```diff
@@ -85,100 +100,92 @@ export default function useDataStore() {
 
-          return data.grantRoundCLR;
-        }) as GrantRoundCLR[]
-      )
-    ).reduce((byRound, data) => {
-      byRound[data.grantRound] = data;
-
-      return byRound;
-    }, {} as Record<string, GrantRoundCLR>);
-
-    // Fetch Metadata
-    const grantMetaPtrs = grantsList.map((grant: Grant) => grant.metaPtr);
-    const grantRoundMetaPtrs = grantRoundsList.map((grantRound: GrantRound) => grantRound.metaPtr);
-    void fetchMetaPtrs(grantMetaPtrs, grantMetadata);
-    void fetchMetaPtrs(grantRoundMetaPtrs, grantRoundMetadata);
     // Save off data
     grants.value = grantsList as Grant[];
     grantContributions.value = contributions.contributions as Contribution[];
     grantRounds.value = grantRoundsList as GrantRound[];
-    grantRoundsCLRData.value = grantDataByRound as Record<string, GrantRoundCLR>;
+    grantRoundsDonationToken.value = SUPPORTED_TOKENS_MAPPING[grantRoundDonationTokenAddress] as TokenInfo;
   }
 
   /**
-   * @notice Helper method that fetches metadata for a Grant or GrantRound, and saves the data
-   * to the state as soon as it's received
-   * @param metaPtrs Array of URLs to resolve
-   * @param metadata Name of the store's ref to assign resolve metadata to
-   */
-  async function fetchMetaPtrs(metaPtrs: string[], metadata: Ref) {
-    const newMetadata = metaPtrs
-      .filter((metaPtr) => !metadata.value[metaPtr])
-      .reduce((prev, cur) => {
-        return {
-          ...prev,
-          [cur]: { status: 'pending' },
-        };
-      }, {});
-    // save these pending metadata objects to state
-    metadata.value = { ...metadata.value, ...newMetadata };
-    // resolve metadata via metaPtr and update state
-    void (await Promise.all(
-      Object.keys(newMetadata).map(async (url) => {
-        try {
-          // save each individual ipfs result into storage
-          let data = getStorage('ipfs-' + url);
-          if (!data) {
-            data = await resolveMetaPtr(url);
-            setStorage('ipfs-' + url, data);
-          }
-          metadata.value[url] = { status: 'resolved', ...data };
-        } catch (e) {
-          metadata.value[url] = { status: 'error' };
-          console.error(e);
-        }
-      })
-    ));
-    // trigger a set at the root
-    metadata.value = { ...metadata.value };
-
-    return metadata;
-  }
-
-  /**
-   * @notice Throttle calls to rawPoll()
+   * @notice Throttle calls to rawPoll() and catch any errors
    */
   let timeout: ReturnType<typeof setTimeout> | undefined;
-  function poll(forceRefresh = false) {
+  async function poll(forceRefresh = false) {
     if (!timeout) {
-      rawPoll(forceRefresh);
-      timeout = setTimeout(function () {
-        timeout = undefined;
-      }, 1000);
+      try {
+        await rawPoll(forceRefresh);
+        timeout = setTimeout(function () {
+          timeout = undefined;
+        }, 1000);
+      } catch (e) {
+        console.log('dGrants: Data fetch error - please check your network -- ', e);
```

Comment:
Do we want to throw the error after catching it/logging it, that way it surfaces to the user as an alert in the top right? Or maybe we don't want to show this, depending on when you saw this and what prompted you to add it. Also worth mentioning another option is to add the error message to the `messagesToIgnore` array

```suggestion
        console.log('dGrants: Data fetch error - please check your network -- ', e);
        throw e;
```

Reply:
I've removed this try/catch! 🙌